minvws/nl-kat-coordination
# Add search endpoint for schedules for scheduler

### Changes

Added `/schedules/search` endpoint to the scheduler api

### QA notes

Added an extra endpoint to the scheduler that allows more complex queries to be made to retrieve schedules.

1. Go through onboarding
2. Start a scan
3. Going to query the `/schedules/search` to check if we can use more complex queries to get a schedule out
4. 
```console
curl -X POST http://localhost:8004/schedules/search -H "Content-Type: application/json" -d \
'{
  "filters": [
    {
      "column": "data",
      "field": "boefje__id",
      "operator": "eq",
      "value": "dns-zone"
    }
  ]
}'
```
5. Should yield a schedule, if it doesn't try a different boefje id
---

### Code Checklist







---

## Checklist for code reviewers:

Copy-paste the checklist from [the docs/source/templates folder](https://github.com/minvws/nl-kat-coordination/blob/main/docs/source/templates/pull_request_template_review_code.md) into your comment.

---

## Checklist for QA:

Copy-paste the checklist from [the docs/source/templates folder](https://github.com/minvws/nl-kat-coordination/blob/main/docs/source/templates/pull_request_template_review_qa.md) into your comment.
